File: core/src/main/java/com/mcreater/amclcore/model/oauth/DeviceCodeConverterModel.java
package com.mcreater.amclcore.model.oauth;

import com.mcreater.amclcore.annotations.OAuthLoginModel;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
@OAuthLoginModel
public class DeviceCodeConverterModel {
    private TokenResponseModel model;
    private boolean isDevice;

    public String createAccessToken() {
        return isDevice() ? "d=" + getModel().getAccessToken() : getModel().getAccessToken();
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/DeviceCodeConverterModel.kt
package com.mcreater.amclcore.model.oauth

data class DeviceCodeConverterModel(
    var model: TokenResponseModel? = null,
    var isDevice: Boolean = false
) {
    fun createAccessToken(): String {
        return if (isDevice) "d=" + (model?.accessToken ?: "") else (model?.accessToken ?: "")
    }
}
